aten/src/ATen/core/type.cpp

```diff
@@ -560,8 +570,12 @@ TupleType::TupleType(
 }
 
 bool TupleType::isSubtypeOfExt(const TypePtr rhs_, std::ostream* why_not) const {
-  if (Type::isSubtypeOfExt(rhs_, why_not))
+  if (Type::isSubtypeOfExt(rhs_, why_not)) {
+    return true;
+  }
+  if (rhs_->kind() == AnyTupleType::Kind) {
     return true;
+  }
   auto rhs = rhs_->cast<TupleType>();
   if (!rhs)
```

Comment:
If this test is isomorphic to the new one, we should probably standardize on one and kill the other (or maybe leave it as an assert, I guess)

...and if there's a motivation for leaving them both in, maybe worth noting it in a comment, as it's not super obvious.

Reply:
AnyTupleType doesn't cast to TupleType. It would be hard to make that work since TupleType is also expected to list its members. I think this is what we want for now because use of AnyTupleType is constrained only to isinstance checks.